Constrain bond direction in ``add_structure_coord`` function for constraints

hdsassnick opened the pull request:
In the previous version the bond direction would be altered when ``dist_constraints``, leading to unreasonable configurations.

This PR makes sure that the bond direction as determined via the coordination is maintained and adds a test to ensure appropriate behaviour.

Scemp reviewed (commented):
It looks good to me, but I was wondering why we use three rotations instead of two to cover the sphere?

hdsassnick commented on aim2dat/strct/ext_manipulation/add_structure.py:
```diff
@@ -287,9 +288,9 @@ def add_structure_coord(
-        for alpha in np.linspace(-0.5 * np.pi, 0.5 * np.pi, num=5):
-            for beta in np.linspace(-0.5 * np.pi, 0.5 * np.pi, num=5):
-                for gamma in np.linspace(0.0, 2.0 * np.pi, num=10):
+        for alpha in np.linspace(0.0, 2.0 * np.pi, num=10):
+            for beta in np.linspace(-1.0 * np.pi, 1.0 * np.pi, num=10):
+                for gamma in np.linspace(-1.0 * np.pi, 1.0 * np.pi, num=10):
```
Initially I tried it with 2 angles, but I couldn't reach appropriate orientations of the molecule in all cases.

hdsassnick commented:
@Scemp, as discussed, I fixed the corner cases and implemented an additional validation check in the ``Structure`` class to exclude ``nan`` values for the positions.

If there are no other objections, this should be ready to be merged..
